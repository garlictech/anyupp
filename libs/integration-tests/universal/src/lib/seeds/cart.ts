/* eslint-disable @typescript-eslint/no-non-null-assertion */
import { cartSeed } from '../fixtures/cart';
import { tap, catchError } from 'rxjs/operators';
import { throwError } from 'rxjs';
import {
  crudBackendGraphQLClient,
  executeMutation,
} from '@bgap/shared/graphql/api-client';
import { CrudApi, CrudApiMutationDocuments } from '@bgap/crud-gql/api';
import { resultTap } from './seed.util';

export const createTestCart = (input: Partial<CrudApi.CreateCartInput>) =>
  executeMutation(crudBackendGraphQLClient)<CrudApi.CreateCartMutation>(
    CrudApiMutationDocuments.createCart,
    { input },
  ).pipe(resultTap('CART create', input.id!));

export const deleteTestCart = (id: string) =>
  executeMutation(crudBackendGraphQLClient)<CrudApi.DeleteCartMutation>(
    CrudApiMutationDocuments.deleteCart,
    {
      input: { id },
    },
  ).pipe(resultTap('CART delete', id));
