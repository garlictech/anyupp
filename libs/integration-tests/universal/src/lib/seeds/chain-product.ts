import { CrudSdk } from '@bgap/crud-gql/api';
import { CreateChainProductInput } from '@bgap/domain';
import { resultTap } from './seed.util';

export const createTestChainProduct = (
  input: CreateChainProductInput,
  crudSdk: CrudSdk,
) =>
  crudSdk
    .CreateChainProduct({ input })
    .pipe(resultTap('CHAIN PRODUCT create', input.id!));

export const deleteTestChainProduct = (id: string, crudSdk: CrudSdk) =>
  crudSdk
    .DeleteChainProduct({ input: { id } })
    .pipe(resultTap('CHAIN PRODUCT delete', id));
