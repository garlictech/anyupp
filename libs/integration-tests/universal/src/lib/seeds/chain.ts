import * as CrudApi from '@bgap/crud-gql/api';
import { resultTap } from './seed.util';

export const createTestChain = (
  input: CrudApi.CreateChainInput,
  crudSdk: CrudApi.CrudSdk,
) => crudSdk.CreateChain({ input }).pipe(resultTap('CHAIN create', input.id!));

export const deleteTestChain = (id: string, crudSdk: CrudApi.CrudSdk) =>
  crudSdk.DeleteChain({ input: { id } }).pipe(resultTap('CHAIN delete', id));
