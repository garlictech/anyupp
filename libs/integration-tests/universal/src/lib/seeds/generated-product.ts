import { CrudApi, CrudApiMutationDocuments } from '@bgap/crud-gql/api';
import {
  crudBackendGraphQLClient,
  executeMutation,
} from '@bgap/shared/graphql/api-client';
import { of } from 'rxjs';
import { catchError } from 'rxjs/operators';
import { resultTap } from './seed.util';

export const createTestGeneratedProduct = (
  input: CrudApi.CreateGeneratedProductInput,
) =>
  executeMutation(crudBackendGraphQLClient)<
    CrudApi.CreateGeneratedProductMutation
  >(CrudApiMutationDocuments.createGeneratedProduct, {
    input,
  }).pipe(resultTap('GENERATED_PRODUCT create', input.id!));

export const deleteTestGeneratedProduct = (id: string) =>
  executeMutation(crudBackendGraphQLClient)<
    CrudApi.DeleteGeneratedProductMutation
  >(CrudApiMutationDocuments.deleteGeneratedProduct, {
    input: { id },
  }).pipe(
    resultTap('GENERATED_PRODUCT delete', id),
    catchError(() => of('NO_PROBLEM')),
  );
