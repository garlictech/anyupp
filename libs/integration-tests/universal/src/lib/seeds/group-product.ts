import { CrudSdk } from '@bgap/crud-gql/api';
import { CreateGroupProductInput } from '@bgap/domain';
import { resultTap } from './seed.util';

export const createTestGroupProduct = (
  input: CreateGroupProductInput,
  crudSdk: CrudSdk,
) =>
  crudSdk
    .CreateGroupProduct({ input })
    .pipe(resultTap('GROUP PRODUCT create', input.id!));

export const deleteTestGroupProduct = (id: string, crudSdk: CrudSdk) =>
  crudSdk
    .DeleteGroupProduct({ input: { id } })
    .pipe(resultTap('GROUP PRODUCT delete', id));
