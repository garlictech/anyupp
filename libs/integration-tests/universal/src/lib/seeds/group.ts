import * as CrudApi from '@bgap/crud-gql/api';
import { resultTap } from './seed.util';

export const createTestGroup = (
  input: CrudApi.CreateGroupInput,
  crudSdk: CrudApi.CrudSdk,
) => crudSdk.CreateGroup({ input }).pipe(resultTap('GROUP create', input.id!));

export const deleteTestGroup = (id: string, crudSdk: CrudApi.CrudSdk) =>
  crudSdk.DeleteGroup({ input: { id } }).pipe(resultTap('GROUP delete', id));
