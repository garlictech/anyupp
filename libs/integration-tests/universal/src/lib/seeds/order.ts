import * as CrudApi from '@bgap/crud-gql/api';

import { resultTap } from './seed.util';

export const createTestOrder = (
  input: CrudApi.CreateOrderInput & { id: string },
  crudSdk: CrudApi.CrudSdk,
) => crudSdk.CreateOrder({ input }).pipe(resultTap('ORDER create', input.id));

export const deleteTestOrder = (id: string, crudSdk: CrudApi.CrudSdk) =>
  crudSdk.DeleteOrder({ input: { id } }).pipe(resultTap('ORDER delete', id));
