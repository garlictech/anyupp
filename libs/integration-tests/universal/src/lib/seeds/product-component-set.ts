import { CrudApi, CrudApiMutationDocuments } from '@bgap/crud-gql/api';
import {
  crudBackendGraphQLClient,
  executeMutation,
} from '@bgap/shared/graphql/api-client';
import { resultTap } from './seed.util';

export const createTestProductComponentSet = (
  input: CrudApi.CreateProductComponentSetInput,
) =>
  executeMutation(crudBackendGraphQLClient)<
    CrudApi.CreateProductComponentSetMutation
  >(CrudApiMutationDocuments.createProductComponentSet, {
    input,
  }).pipe(resultTap('PRODUCT_COMPONENT_SET create', input.id!));

export const deleteTestProductComponentSet = (id: string) =>
  executeMutation(crudBackendGraphQLClient)<
    CrudApi.DeleteProductComponentSetMutation
  >(CrudApiMutationDocuments.deleteProductComponentSet, {
    input: { id },
  }).pipe(resultTap('PRODUCT_COMPONENT_SET delete', id));

export const createTestProductComponent = (
  input: CrudApi.CreateProductComponentInput,
) =>
  executeMutation(crudBackendGraphQLClient)<
    CrudApi.CreateProductComponentMutation
  >(CrudApiMutationDocuments.createProductComponent, {
    input,
  }).pipe(resultTap('PRODUCT_COMPONENT create', input.id!));

export const deleteTestProductComponent = (id: string) =>
  executeMutation(crudBackendGraphQLClient)<
    CrudApi.DeleteProductComponentMutation
  >(CrudApiMutationDocuments.deleteProductComponent, {
    input: { id },
  }).pipe(resultTap('PRODUCT_COMPONENT delete', id));
