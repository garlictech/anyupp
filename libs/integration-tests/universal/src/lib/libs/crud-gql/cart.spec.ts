import * as CrudApi from '@bgap/crud-gql/api';
import {
  cartFixture,
  testAdminEmail,
  testAdminUserPassword,
} from '@bgap/shared/fixtures';
import { switchMap } from 'rxjs/operators';
import { createAuthenticatedCrudSdk } from '../../../api-clients';
import { createTestCart, deleteTestCart } from '../../seeds/cart';

describe('getCart test', () => {
  let authSdk: CrudApi.CrudSdk;

  beforeAll(async () => {
    authSdk = await createAuthenticatedCrudSdk(
      testAdminEmail,
      testAdminUserPassword,
    ).toPromise();

    await deleteTestCart(cartFixture.cart_01.id, authSdk)
      .pipe(switchMap(() => createTestCart(cartFixture.cart_01, authSdk)))
      .toPromise();
  }, 15000);

  afterAll(async () => {
    await deleteTestCart(cartFixture.cart_01.id, authSdk).toPromise();
  });

  it('successful query execution', done => {
    authSdk.GetCart({ id: cartFixture.cart_01.id }).subscribe({
      next(x) {
        expect(x?.id).toEqual(cartFixture.cart_01.id);
        done();
      },
    });
  }, 15000);

  it('should return null for a not existing item', done => {
    authSdk.GetCart({ id: cartFixture.cartId_NotExisting }).subscribe({
      next(x) {
        expect(x).toBeNull();
        done();
      },
    });
  }, 10000);

  it('should throw error without id as input', done => {
    authSdk.GetCart(undefined as any).subscribe({
      error(e) {
        expect(e).toMatchSnapshot();
        done();
      },
    });
  }, 15000);
});
