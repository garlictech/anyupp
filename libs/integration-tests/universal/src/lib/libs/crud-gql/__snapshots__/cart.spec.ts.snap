// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCart test should throw error without id as input 1`] = `[Error: GraphQL error: Variable 'id' has coerced Null value for NonNull type 'ID!']`;
